src/Psl/Fun/pipe.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Psl\Fun;
+
+use function Psl\Iter\reduce;
+
+/**
+ * Performs left-to-right function composition.
+ * Example
+ *      $pipe = Fun\pipe(
+ *          fn($value) => 'Hello' . $value,
+ *          fn($value) => $value . '!',
+ *          fn($value) => '¡' . $value
+ *      );
+ *      => $greet('World') === '¡Hello World!';
+ *      => $greet('Jim') === '¡Hello Jim!';
+ *
+ * @psalm-param list<callable(mixed): mixed> $stages
+ *
+ * @psalm-return callable(mixed): mixed
```

Comment:
hm, i don't really like `mixed` here, what do you think about:

```
@template T

@psalm-param callable(T): T ...$stages

@psalm-return callable(T): T
```

Reply:
Makes sense, but will still be "mixed" internally.
The problem here is that you can't really keep track of all these intermediate callable's Input/Output types.
I'll change it nevertheless.